fabtests/common/shared.c

```diff
@@ -536,14 +536,19 @@ static int ft_alloc_ctx_array(struct ft_context **mr_array, char ***mr_bufs,
 	return 0;
 }
 
-static void ft_set_tx_rx_sizes(size_t *set_tx, size_t *set_rx)
+static void ft_set_tx_rx_sizes(size_t *set_tx, size_t *set_rx, size_t *set_rx_msg_size)
 {
 	*set_tx = opts.options & FT_OPT_SIZE ?
 		  opts.transfer_size : test_size[TEST_CNT - 1].size;
 	if (*set_tx > fi->ep_attr->max_msg_size)
 		*set_tx = fi->ep_attr->max_msg_size;
 	*set_rx = *set_tx + ft_rx_prefix_size();
 	*set_tx += ft_tx_prefix_size();
+	*set_rx_msg_size = *set_rx;
+	if (ep)
+		(void) fi_getopt(&ep->fid, FI_OPT_ENDPOINT, FI_OPT_MAX_MSG_SIZE,
+				 set_rx_msg_size,
+				 &(size_t) {sizeof *set_rx_msg_size});
```

Comment:
This is incomplete.  It saves a new rx_msg_size, which may be smaller than rx_size, but doesn't adjust the tx_size?  That's just going to result in a buffer overrun.  The change only targets msg transfers, not tagged.

If the intent is to have per transfer type size limits, I would make this more generic and maybe rework this area of the code.

That this change requires a check for an ep suggests that the ep may not always be available, which restricts how well this change can be supported.